fast-af/fast-checkout-woocommerce
# PINT-331 - Add partial refunded order support

# Description

Currently, order-consistency-job doesn't detect missing partial refunds for WC and would be good to have this as we put more dependency on this cronjob.

WC inherently doesn't have an API to fetch partial refunds since there is no concept/status of PARTIAL_REFUNDED on WC platform. However, we can use our existing WC Plugin and expose a new endpoint to expose this API.

See https://fastaf.atlassian.net/browse/PINT-331 for more info.

# Testing instructions

1. Using PostMan, hit this REST API endpoint: https://fasttestdev.wpcomstaging.com/wp-json/wc/fast/v1/refunds
2. Use the available payload parameters described here: https://fastaf.atlassian.net/browse/PINT-331?focusedCommentId=26371
3. Verify that the API returns refunds as expected.

# Checks

@ilkerulutas @brikr this is ready for code review.
